clustered/client/src/test/java/org/ehcache/clustered/client/internal/store/AbstractServerStoreProxyTest.java

```diff
@@ -0,0 +1,80 @@
+ * limitations under the License.
+ */
+package org.ehcache.clustered.client.internal.store;
+
+import org.ehcache.clustered.client.internal.ClusterTierManagerClientEntityFactory;
+import org.ehcache.clustered.client.internal.ClusterTierManagerClientEntityService;
+import org.ehcache.clustered.client.internal.UnitTestConnectionService;
+import org.ehcache.clustered.client.internal.UnitTestConnectionService.PassthroughServerBuilder;
+import org.ehcache.clustered.client.internal.lock.VoltronReadWriteLockEntityClientService;
+import org.ehcache.clustered.common.ServerSideConfiguration;
+import org.ehcache.clustered.common.internal.ServerStoreConfiguration;
+import org.ehcache.clustered.lock.server.VoltronReadWriteLockServerEntityService;
+import org.ehcache.clustered.server.ClusterTierManagerServerEntityService;
+import org.ehcache.clustered.server.store.ObservableClusterTierServerEntityService;
+import org.ehcache.config.units.MemoryUnit;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.terracotta.connection.Connection;
+
+import java.net.URI;
+import java.util.Collections;
+import java.util.Properties;
+
+public abstract class AbstractServerStoreProxyTest {
+
+  private static final URI CLUSTER_URI = URI.create("terracotta://localhost");
+  private static final UnitTestConnectionService CONNECTION_SERVICE = new UnitTestConnectionService();
+
+  protected static ObservableClusterTierServerEntityService OBSERVABLE_CLUSTER_TIER_SERVICE;
+
+  @BeforeClass
+  public static void createCluster() {
+    UnitTestConnectionService.add(CLUSTER_URI, new PassthroughServerBuilder()
+      .serverEntityService(new ClusterTierManagerServerEntityService())
+      .clientEntityService(new ClusterTierManagerClientEntityService())
+      .serverEntityService(OBSERVABLE_CLUSTER_TIER_SERVICE = new ObservableClusterTierServerEntityService())
+      .clientEntityService(new ClusterTierClientEntityService())
+      .serverEntityService(new VoltronReadWriteLockServerEntityService())
+      .clientEntityService(new VoltronReadWriteLockEntityClientService())
+
+      .resource("defaultResource", 128, MemoryUnit.MB).build());
+  }
+
+  @AfterClass
+  public static void destroyCluster() {
+    UnitTestConnectionService.remove(CLUSTER_URI);
+    OBSERVABLE_CLUSTER_TIER_SERVICE = null;
+  }
+
+  protected static SimpleClusterTierClientEntity createClientEntity(String name,
+                                                                  ServerStoreConfiguration configuration,
+                                                                  boolean create) throws Exception {
+    Connection connection = CONNECTION_SERVICE.connect(CLUSTER_URI, new Properties());
+
+    ClusterTierManagerClientEntityFactory entityFactory = new ClusterTierManagerClientEntityFactory(connection);
+    if (create) {
+      entityFactory.create(name, new ServerSideConfiguration("defaultResource", Collections.<String, ServerSideConfiguration.Pool>emptyMap()));
```

Comment:
The typing was requested because of some inference issue?

Reply:
I would guess so... but now it seems to be okay.

Reply:
Possibly a "Java 6 not good at inferring" thing